### ISSUE TITLE
Databricks dialect does not support VOLUME syntaxes

### ISSUE THREAD

**Issue by TheCleric** [sqlfluff/sqlfluff#6178]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting sql in the Databricks dialect that has commands involving VOLUMEs, sqlfluff gives an "unparsable section" error.

### Expected Behaviour

sqlfluff will correctly lint commands involving VOLUMEs.

See: https://docs.databricks.com/en/sql/language-manual/sql-ref-volumes.html

### Observed Behaviour

An error like the following is generated:
`Found unparsable section: 'CREATE VOLUME IF NOT EXISTS bronze.track...'

### How to reproduce

Attempt to lint the following. It is known valid and works in Databricks

create_alter_volume.sql
```sql
CREATE VOLUME IF NOT EXISTS bronze.tracking;
ALTER VOLUME bronze.tracking SET OWNER TO `admins`;
```

### Dialect

Databricks

### Version

sqlfluff, version 3.1.1
Python 3.10.13

### Configuration

.sqlfluff
```toml
[sqlfluff]
verbose = 2
dialect = databricks
templater = placeholder
max_line_length = 120
exclude_rules = CP02, ST06
warnings = RF03
large_file_skip_byte_limit = 0

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = True
indented_ctes = True
indented_using_on = True
indented_on_contents = True
# indented_then = True
# indented_then_contents = True
allow_implicit_indents = True
template_blocks_indent = True

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
